pyresample/geometry.py

```diff
@@ -729,10 +726,11 @@ def freeze(self, lonslats=None,
             yarr[yarr > 9e29] = np.nan
             corners = [np.nanmin(xarr), np.nanmin(yarr),
                        np.nanmax(xarr), np.nanmax(yarr)]
-
-            domain = self.compute_domain(corners, resolution, size)
+            # For shape and area_extent to be defined, resolution or size must be defined.
+            domain = None
+            if not (resolution is None and size is None):
+                domain = self.compute_domain(corners, resolution, size)
             self.area_extent, self.x_size, self.y_size = domain
```

Comment:
Won't this line fail if either of them is specified? Maybe `compute_domain` should check for neither of them to be specified and for both of them specified (as it is doing already)?